docs/important/relevant_functionalities_de.md

```diff
@@ -0,0 +1,1250 @@
+
+Das Post-Upgrade-Skript kann als [Exposed Command](../core/compendium_de.md#exposedcommands) in der dogu.json eines Dogus definiert werden:
+
+```json
+  "ExposedCommands": [
+    {
+      "Name": "post-upgrade",
+      "Command": "/post-upgrade.sh"
+    }
+  ]
+```
+
+Dieses Skript wird nach dem Upgrade des Dogus im neuen Dogu-Container gestartet, sobald dieser hochfährt. Es laufen also im Normalfall sowohl das `startup.sh`- als auch das `post-upgrade.sh`-Skript gleichzeitig los. Falls es notwendig sein sollte, dass das `post-upgrade.sh`-Skript noch vor dem `startup.sh`-Skript ausgeführt wird, muss das `startup.sh`-Skript mit einem Wartemechanismus versehen werden. Dies kann bspw. durch das Warten auf einen etcd-Key geschehen, den das `post-upgrade.sh`-Skript setzt, sobald es komplett durchgelaufen ist.
+
+### upgrade-notification.sh - Zeigt eine Benachrichtigung vor der Upgradebestätigung eines Dogus
+
+Das Upgrade-Notification-Skript kann als [Exposed Command](../core/compendium_de.md#exposedcommands) in der dogu.json eines Dogus definiert werden:
+
+```json
+  "ExposedCommands": [
+    {
+      "Name": "upgrade-notification",
+      "Command": "/upgrade-notification.sh"
+    }
+  ]
+```
+
+Dieses Skript wird vor dem Upgrade-Vorgang ausgeführt und sollte ausschließlich Informationen (per `echo`) ausgeben, die für den Administrator vor dem Upgrade relevant sein können. Beispielsweise kann hier auf Breaking Changes hingewiesen werden oder darauf, dass man vor dem Upgrade ein Backup anlegen sollte.
+
+
+## Typische Dogu-Features
+
+Dieses Kapitel beschreibt Funktionen, die Dogus tiefer in das Cloudogu EcoSystem integrieren und sie einheitlich administrierbar machen.
+
+### Memory-/Swap-Limit
+
+Mit Memory- und Swap-Limits kann man den Speicherverbrauch (Arbeitsspeicher und Swap) von Dogus einschränken.
+
+Falls ein Dogu seine Speicherlimitierung überschreitet, wird der größte Prozess im Container beendet.
+Dies ist normalerweise der Hauptprozess des Dogus und führt dazu, dass der Container neu gestartet wird.
+
+Wird kein Wert bei der Memory-Limitierung gesetzt, findet diese auch nicht statt.
+Bei der Swap-Limitierung is `0b` der Standardwert und stellt somit keinen Swap zur Verfügung.
+
+Um eine Limitierung vornehmen zu können, muss die dogu.json des Dogus folgende Einträge enthalten:
+
+```json
+{
+    "Name": "container_config/memory_limit",
+    "Description":"Limits the container's memory usage. Use a positive integer value followed by one of these units [b,k,m,g] (byte, kibibyte, mebibyte, gibibyte).",
+    "Optional": true,
+    "Validation": {
+    "Type": "BINARY_MEASUREMENT"
+    }
+},
+{
+    "Name": "container_config/swap_limit",
+    "Description":"Limits the container's swap memory usage. Use zero or a positive integer value followed by one of these units [b,k,m,g] (byte, kibibyte, mebibyte, gibibyte). 0 will disable swapping.",
+    "Optional": true,
+    "Validation": {
+    "Type": "BINARY_MEASUREMENT"
+    }
+}
```

Comment:
Hinting the full json format would have prevented weird indentation. I suggest to add outer braces like so
```suggestion
{
...,
  {
    "Name": "container_config/memory_limit",
    "Description":"Limits the container's memory usage. Use a positive integer value followed by one of these units [b,k,m,g] (byte, kibibyte, mebibyte, gibibyte).",
    "Optional": true,
    "Validation": {
    "Type": "BINARY_MEASUREMENT"
    }
},
{
    "Name": "container_config/swap_limit",
    "Description":"Limits the container's swap memory usage. Use zero or a positive integer value followed by one of these units [b,k,m,g] (byte, kibibyte, mebibyte, gibibyte). 0 will disable swapping.",
    "Optional": true,
    "Validation": {
    "Type": "BINARY_MEASUREMENT"
    }
  }
  ...,
}
```